packages/rollup-core/test/app/block-batch-processor.spec.ts

```diff
@@ -0,0 +1,652 @@
+        1,
+        'Should have received a block batch'
+      )
+      listener.receivedBlockBatches[0].timestamp.should.equal(
+        timestamp,
+        'Timestamp mismatch'
+      )
+      listener.receivedBlockBatches[0].blockNumber.should.equal(
+        blockNumber,
+        'Block number mismatch'
+      )
+      listener.receivedBlockBatches[0].batches.length.should.equal(
+        2,
+        'Num batches mismatch'
+      )
+
+      listener.receivedBlockBatches[0].batches[0].length.should.equal(
+        1,
+        'Transaction count mismatch'
+      )
+      listener.receivedBlockBatches[0].batches[1].length.should.equal(
+        1,
+        'Transaction count mismatch'
+      )
+
+      rollupTxsEqual(
+        listener.receivedBlockBatches[0].batches[0][0],
+        rollupTx
+      ).should.eq(true, 'tx mismatch')
+      rollupTxsEqual(
+        listener.receivedBlockBatches[0].batches[1][0],
+        rollupTx2
+      ).should.eq(true, 'tx2 mismatch')
+    })
+
+    it('should handle block with multiple logs on different topics as separate batches', async () => {
+      const timestamp = 123
+      const blockNumber = 0
+      const txHash = getHashFromString('derp derp derp')
+      calldataParseMap.set(calldata, [rollupTx])
+
+      mockedLogsProvider.transactionsByHash.set(
+        txHash,
+        getTransactionResponse(timestamp, calldata, txHash)
+      )
+
+      const txHash2 = getHashFromString('derp derp derp2')
+      calldataParseMap.set(calldata2, [rollupTx2])
+
+      mockedLogsProvider.transactionsByHash.set(
+        txHash2,
+        getTransactionResponse(timestamp, calldata2, txHash2)
+      )
+
+      mockedLogsProvider.logsToReturn.push([
+        getLog([singleTxTopic], singleTxContractAddress, txHash, 1),
+        getLog([batchTxTopic], batchTxContractAddress, txHash2, 2),
+      ])
+
+      await blockBatchProcessor.handle(getBlock(timestamp, blockNumber))
+
+      await sleep(100)
+
+      listener.receivedBlockBatches.length.should.equal(
+        1,
+        'Should have received a block batche'
+      )
+      listener.receivedBlockBatches[0].timestamp.should.equal(
+        timestamp,
+        'Timestamp mismatch'
+      )
+      listener.receivedBlockBatches[0].blockNumber.should.equal(
+        blockNumber,
+        'Block number mismatch'
+      )
+      listener.receivedBlockBatches[0].batches.length.should.equal(
+        2,
+        'Num batches mismatch'
+      )
+
+      listener.receivedBlockBatches[0].batches[0].length.should.equal(
+        1,
+        'First batch tx count mismatch'
+      )
+      listener.receivedBlockBatches[0].batches[1].length.should.equal(
+        1,
+        'Second batch tx count mismatch'
+      )
+
+      rollupTxsEqual(
+        listener.receivedBlockBatches[0].batches[0][0],
+        rollupTx
+      ).should.eq(true, 'tx mismatch')
+      rollupTxsEqual(
+        listener.receivedBlockBatches[0].batches[1][0],
```

Comment:
Nice ok so this is called a `BlockBatch` and now we have each normal batch just called batch? ok that works & sounds good!

Reply:
Yep. `BlockBatches` is the type. Weird to be plural, I know, but indicates that it contains all the batches from a block.